### PR TITLE
Migrate OIDC, OIDC client and OIDC Client Registration extensions to @ConfigMapping

### PR DESCRIPTION
**Pull request by michalvavrik**
- implements the first phase of agreed plan https://github.com/quarkusio/quarkus/issues/39185#issuecomment-2439669375: migrates config classes to `@ConfigMapping`

Migration of OIDC configuration classes to the `@ConfigMapping` is complicated because:

- we modify config class `@ConfigGroup` instances, we change values after runtime config setup has finished
- we use the config class `@ConfigGroup` as part of "public API" (meaning: they are in documentation, users directly produce instances of these classes and work with them)
- abstraction ([ `OidcTenantConfig` || `OidcClientConfig` ] -> `OidcClientCommonConfig` -> `OidcCommonConfig` and `OidcClientRegistrationConfig` -> `OidcCommonConfig`) shared by 3 extensions
- some OIDC configs are really huge (like OidcTenantConfig)

This PR introduces `@ConfigMapping` interfaces and makes original config classes POJOs.
These POJOs are still used everywhere by Quarkus OIDC and users.

Regarding affected Quarkiverse extensions:

- Renarde will need to add `()`:
  -  https://github.com/quarkiverse/quarkus-renarde/blob/main/oidc/src/main/java/io/quarkiverse/renarde/oidc/impl/RenardeTenantResolver.java#L32
  - https://github.com/quarkiverse/quarkus-renarde/blob/e0dc19318068744e276fdd4535ac8e09c48b43d0/oidc/src/main/java/io/quarkiverse/renarde/oidc/impl/RenardeOidcTenantProvider.java#L28

### CODE REVIEW

**Comment by sberyozkin** [comment]
Awesome, thanks @michalvavrik, I'll start reviewing asap, thanks very much

**Comment by michalvavrik** [comment]
> Awesome, thanks @michalvavrik, I'll start reviewing asap, thanks very much

Yeah, I don't want to hurry you as I know you are busy, but there is one thing: I checked all OIDC PRs between I started working on this and now and there were no relevant change, but if anything that touches OIDC config changes, this PR needs to be updated. So we need to sync with changes in main repo (hopefully it would result in a merge conflict, but I am not 100 % certain).

**Comment by michalvavrik** [comment]
I have rebased no current main, adapted this PR to changes made in  #44114 (AKA added Slack) and checked PRs this week so far, no relevant changes. It's safe so far. I'll re-check in a few days as well. No hurry.

**Comment by sberyozkin** [comment]
@michalvavrik Michal, can you type some DSL here ? Basic ones like the clientid, secret, authserver url and web-app application type only, and more complex ones, where in addition to the basics, users setup set up some `authentication` properties, and use JWT private key client authentication... For the latter part, it would be great to have an option to do all not only in a single sequence, but create individual parts and then assemble together...

**Comment by sberyozkin** [comment]
Sorry @michalvavrik, I forgot that phase 1 is only about supporting `@ConfigMapping`

File: extensions/oidc/runtime/src/test/java/io/quarkus/oidc/runtime/OidcTenantConfigImpl.java [hunk 1]
```diff
@@ -0,0 +1,1238 @@
...
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.configuration.TrimmedStringConverter;
+import io.smallrye.config.WithConverter;
+
+final class OidcTenantConfigImpl implements OidcTenantConfig {
```
Comment (sberyozkin) [inline extensions/oidc/runtime/src/test/java/io/quarkus/oidc/runtime/OidcTenantConfigImpl.java:14]: @michalvavrik Can you clarify please what does this class do ?

Comment (michalvavrik) [reply extensions/oidc/runtime/src/test/java/io/quarkus/oidc/runtime/OidcTenantConfigImpl.java:14]: It is a test class that I use to test that mapper between a `@ConfigMapping` interface and former `@ConfigGroup` class maps every single interface method. If I do:

```
configGroupClass.nameOne = interfaceGroup.nameOne();
```
then when `nameOne` is invoked, at least I know that it wasn't forgotten.

**Comment by michalvavrik** [comment]
FYI - I just rebased and checked commits. Looks like nothing relevant changed so I didn't have to do anything.

File: extensions/oidc/runtime/src/test/java/io/quarkus/oidc/runtime/OidcTenantConfigImpl.java (hunk 1 above)
Comment (sberyozkin) [reply extensions/oidc/runtime/src/test/java/io/quarkus/oidc/runtime/OidcTenantConfigImpl.java:14]: Oh sorry, it is a test class

**Review by sberyozkin — APPROVED** [review]
Let's start the migration process, thanks @michalvavrik

**Comment by michalvavrik** [comment]
Thank you @sberyozkin , I'll open Renarde PR to address this change. (these 2 bracket scenarios)

**Pull request merged**